### DISCUSSION TITLE
Pocket SNES Cart RW - Clock generator not working!

### DISCUSSION THREAD

**Discussion by Hanzel1985** [sanni/cartreader#390] [Need help]
for some reason, with the new update, i can enable the adafruit clock generator, even if i configure in the main option part in the arduino programmer, can someone help¡

**Comment by sanni** [comment]
Check that the SDA and SCL pins of the Clock Generator are connected to the Arduino.
The newer Cart_Reader sketch versions check if the Clock Gen is actually working and print an error if it doesn't.

**Comment by Hanzel1985** [comment]
thanks for the reply sanni!, i see, but where i need to connect those pins??, im using the snes pocket cart rw that uses the arduino mega 2560 pro,

**Comment by sanni** [comment]
I don't know how the Pocket SNES Cart RW is wired as it is not my design. Can you make a picture how the Clock Generator is connected right now?

Generally speaking, it needs to be connected like this:
Clock Gen -> Arduino
VIN -> 5V
GND -> GND
SDA -> D20
SCL -> D21

**Reply by Niltonn** [reply]
I'm sorry for the little information, I designed this pcb without even creating a schematic, although it still has been popularized in Brazil, I imagine it's mainly because of the low cost.
As soon as I have more time, I intend to improve this version and make the files available following the guidelines on Sanni's wiki
https://github.com/sanni/cartreader/wiki/Designing-your-own-Adapters
Well, about the connections, I followed the original pinout of the sanni card reader, these connections contemplating the bus of a 46p snes cartridge (without connections with cic) and on the pcb an extra pin outside the bus of the 46p slot that can be connected to the pin 1 of a 62p cartridge, to provide a clock signal, and I used the "stand alone NP.ino" sketch for a while
Anyway, when updating this small version, this information/connections became clearer.
Thanks for the project and work spent over the years on cart reader :)
Att, Nilton
